ccd-definition/CaseEventToComplexTypes/create-c21/C21Order-gatekeeping.json

```diff
@@ -0,0 +1,84 @@
+[
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "C21OrderComplexType",
+    "CaseEventID": "createC21OrderGatekeeping",
+    "CaseFieldID": "c21Order",
+    "ListElementCode": "orderTitle",
+    "EventElementLabel": "Order title",
+    "FieldDisplayOrder": 1,
+    "DisplayContext": "OPTIONAL"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "C21OrderComplexType",
+    "CaseEventID": "createC21OrderGatekeeping",
+    "CaseFieldID": "c21Order",
+    "ListElementCode": "orderDetails",
+    "EventElementLabel": "Order details",
+    "FieldDisplayOrder": 2,
+    "DisplayContext": "MANDATORY"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "C21OrderComplexType",
+    "CaseEventID": "createC21OrderGatekeeping",
+    "CaseFieldID": "c21Order",
+    "ListElementCode": "document",
+    "EventElementLabel": "Order document",
+    "FieldDisplayOrder": 3,
+    "DisplayContext": "READONLY",
+    "FieldShowCondition": "document=\"DO_NOT_SHOW\""
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "C21OrderComplexType",
+    "CaseEventID": "createC21OrderGatekeeping",
+    "CaseFieldID": "c21Order",
+    "ListElementCode": "orderDate",
+    "EventElementLabel": "Date and time of upload",
+    "FieldDisplayOrder": 4,
+    "DisplayContext": "READONLY",
+    "FieldShowCondition": "document=\"DO_NOT_SHOW\""
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "JudgeAndLegalAdvisorComplexType",
+    "CaseEventID": "createC21OrderGatekeeping",
+    "CaseFieldID": "judgeAndLegalAdvisor",
+    "ListElementCode": "judgeTitle",
+    "EventElementLabel": "Judge or magistrate's title",
+    "FieldDisplayOrder": 1,
+    "DisplayContext": "OPTIONAL"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "JudgeAndLegalAdvisorComplexType",
+    "CaseEventID": "createC21OrderGatekeeping",
+    "CaseFieldID": "judgeAndLegalAdvisor",
+    "ListElementCode": "judgeLastName",
+    "EventElementLabel": "Last name",
+    "FieldDisplayOrder": 2,
+    "DisplayContext": "MANDATORY"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "JudgeAndLegalAdvisorComplexType",
+    "CaseEventID": "createC21OrderGatekeeping",
+    "CaseFieldID": "judgeAndLegalAdvisor",
+    "ListElementCode": "judgeFullName",
+    "EventElementLabel": "Full name",
+    "FieldDisplayOrder": 3,
+    "DisplayContext": "MANDATORY"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "JudgeAndLegalAdvisorComplexType",
+    "CaseEventID": "createC21OrderGatekeeping",
+    "CaseFieldID": "judgeAndLegalAdvisor",
+    "ListElementCode": "legalAdvisorName",
+    "EventElementLabel": "Legal advisor's full name",
+    "FieldDisplayOrder": 4,
+    "DisplayContext": "OPTIONAL"
```

Comment:
does this need to be set in caseEventToComplexTypes? Surely can just set in caseEventToFields. Only the c21 Order needs to be displayContext COMPLEX

Reply:
ah unless this is because legal advisor needs to be optional? and the rest mandatory...

Reply:
The names are mandatory (but the names are only shown when a title is selected. Once a title is selected, the last name/full name must be filled in - but initially everything is optional and the user can click continue straight away